lib/ruby_smb/smb2/tree.rb

```diff
@@ -186,11 +186,22 @@ def open_directory_packet(directory: nil, disposition: RubySMB::Dispositions::FI
         create_request.create_options.directory_file  = 1
         create_request.file_attributes.directory      = 1
         create_request.desired_access.list            = 1
-        create_request.share_access.read_access       = 1 if read
-        create_request.share_access.write_access      = 1 if write
-        create_request.share_access.delete_access     = 1 if delete
         create_request.create_disposition             = disposition
 
+        if read
```

Comment:
I'll defer to @cdelafuente-r7 who might have more insights here on the ramifications of these changes 😄